### PR TITLE
NFDIV 4000 Remove General Order event to caseworker super user

### PR DESCRIPTION
**Pull request by pallavijustice**
### Change description ###

Provide a 'Remove General Order event' to caseworker super user

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-4000

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveGeneralOrder.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_REMOVE_GO)
+            .forStates(POST_SUBMISSION_STATES)
+            .name("Remove General Order")
+            .description("Remove General Order")
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .showEventNotes()
+            .grant(CREATE_READ_UPDATE_DELETE, SUPER_USER)
+            .grantHistoryOnly(CASE_WORKER))
+            .page("removeGeneralOrders")
```
Comment (adamg-hmcts) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveGeneralOrder.java:51]: For the event history, Fubara might also be able to confirm who should see that. We make CaseworkerRemoveScannedDocument and CaseworkerCreateGeneralOrder visible to most roles:
https://github.com/hmcts/nfdiv-case-api/blob/edb768814677050c267714d9eef2601ba90a0c07/src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerCreateGeneralOrder.java#L73

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveGeneralOrder.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_REMOVE_GO)
+            .forStates(POST_SUBMISSION_STATES)
+            .name("Remove General Order")
+            .description("Remove General Order")
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .showEventNotes()
+            .grant(CREATE_READ_UPDATE_DELETE, SUPER_USER)
```
[the remark this answers is not in the document]
Comment (pallavijustice) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveGeneralOrder.java:49]: Confirmed that event should only be available to super user

File: src/test/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveGeneralOrderTest.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+        caseworkerRemoveGO.configure(configBuilder);
+
+        assertThat(getEventsFrom(configBuilder).values())
+            .extracting(Event::getId)
+            .contains(CaseworkerRemoveGeneralOrder.CASEWORKER_REMOVE_GO);
+    }
+
+    @Test
+    void shouldRemoveDocumentForDeletedGeneralOrder() {
```
[the remark this answers is not in the document]
Comment (pallavijustice) [reply src/test/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveGeneralOrderTest.java:65]: Added additional test cases for other scenarios

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveGeneralOrder.java (hunk 1 above)
Comment (pallavijustice) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveGeneralOrder.java:51]: Checked with Fubara and Isabel that history should be available to super user and caseworker

**Review by adamg-hmcts — APPROVED** [review]

**Pull request merged**